rust/lance-table/src/io/commit/external_manifest.rs

```diff
@@ -141,33 +145,60 @@ impl ExternalManifestCommitHandler {
     /// by any number of readers or writers, so care should be taken to ensure
     /// that the manifest is not lost nor any errors occur due to duplicate
     /// operations.
+    #[allow(clippy::too_many_arguments)]
     async fn finalize_manifest(
         &self,
         base_path: &Path,
         staging_manifest_path: &Path,
         version: u64,
         size: u64,
+        e_tag: Option<String>,
         store: &dyn OSObjectStore,
         naming_scheme: ManifestNamingScheme,
-    ) -> std::result::Result<Path, Error> {
+    ) -> std::result::Result<ManifestLocation, Error> {
         // step 1: copy the manifest to the final location
         let final_manifest_path = naming_scheme.manifest_path(base_path, version);
-        match store
+
+        let copied = match store
             .copy(staging_manifest_path, &final_manifest_path)
             .await
         {
-            Ok(_) => {}
-            Err(ObjectStoreError::NotFound { .. }) => return Ok(final_manifest_path), // Another writer beat us to it.
+            Ok(_) => true,
+            Err(ObjectStoreError::NotFound { .. }) => false, // Another writer beat us to it.
             Err(e) => return Err(e.into()),
         };
 
+        // On S3, the etag can change if originally was MultipartUpload and later was Copy
+        // https://docs.aws.amazon.com/AmazonS3/latest/API/API_Object.html#AmazonS3-Type-Object-ETag
+        // We generally only do MultipartUpload for > 5MB files, so we can skip this check
```

Comment:
```suggestion
        // We only do MultipartUpload for > 5MB files, so we can skip this check
```

I hope not `generally` if we're going to skip the step :smile: